test/qa/test_meshnet.py

```diff
@@ -668,3 +668,65 @@ def test_set_defaults_when_connected_2nd_set(tech, proto, obfuscated):
     assert "Status: Disconnected" in sh.nordvpn.status()
 
     assert settings.app_has_defaults_settings()
+
+
+def base_test_peer_list(filter_list: list[str] = None) -> None:
+    sh.nordvpn.mesh.peer.list()
+    ssh_client.exec_command("nordvpn mesh peer list")
+
+    peer_list = meshnet.PeerList.from_str(sh.nordvpn.mesh.peer.list())
+    local_hostname = peer_list.get_this_device().hostname
+    remote_hostname = peer_list.get_external_peer().hostname
+
+    if filter_list is not None:
+        routing_allowed = "routing-allowed" in filter_list
+        incoming_traffic_allowed = "incoming-traffic-allowed" in filter_list
+        allows_sending_files = "allows-sending-files" in filter_list
+        allows_routing = "allows-routing" in filter_list
+        allows_incoming_traffic = "allows-incoming-traffic" in filter_list
+
+        meshnet.set_permissions(remote_hostname, routing=routing_allowed, incoming=incoming_traffic_allowed, fileshare=allows_sending_files)
+        ssh_client.meshnet.set_permissions(local_hostname, routing=allows_routing, incoming=allows_incoming_traffic, fileshare=allows_sending_files)
+
+    local_peer_list = meshnet.get_clean_peer_list(sh.nordvpn.mesh.peer.list())
+    local_formed_list = meshnet.PeerList.from_str(local_peer_list).parse_peer_list(filter_list)
+
+    if len(filter_list) != 0:
+        local_peer_list_filtered = meshnet.get_clean_peer_list(str(sh.nordvpn.mesh.peer.list("-f", ",".join(filter_list)))).split("\n")
+
+        assert local_formed_list == local_peer_list_filtered
+    else:
+        assert local_formed_list == local_peer_list.split("\n")
+
+    if "External Peers:" in "\n".join(local_formed_list) and "External Peers:\n[no peers]" not in "\n".join(local_formed_list):
+        assert "Status: connected" in local_formed_list
+
+
+@pytest.mark.parametrize("external", [True, False], ids=lambda value: "external" if value else "")
+@pytest.mark.parametrize("internal", [True, False], ids=lambda value: "internal" if value else "")
+@pytest.mark.parametrize("offline", [True, False], ids=lambda value: "offline" if value else "")
+@pytest.mark.parametrize("online", [True, False], ids=lambda value: "online" if value else "")
+@pytest.mark.flaky(reruns=3, reruns_delay=20)
```

Comment:
do you know why the tests needs to be rerun? what causes the failure? 
Same for the next one.

Reply:
Sometimes status of peer shows up as Disconnected, therefore I in introduced re-runs